doc/preallocation.md

```diff
@@ -0,0 +1,72 @@
+# Data Volume preallocation
+
+This documents describes how to use DataVolume's `preallocation` option.
+
+## Introduction
+
+In order to improve write performance, QEMU provides the ability to preallocate QCOW2 files, used
+as backing for DataVolumes.
+
+There are several methods for preallocation. CDI uses the best one available, depending on the
+underlying file system and device type. It tries to use the OS's `fallocate` call if the filesystem
+supports it and falls back to "full" preallocation for block devices. Preallocation does not depend
+on the source of the DV, i.e. it can be used for import, upload or blank DVs.
+
+See `qemu-img` documentation to learn more about preallocation.
+
+## Preallocation for DataVolume
+
+To preallocate space for a DataVolume, use `preallocation` field in DataVolume's spec:
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: preallocated-datavolume
+spec:
+  source:
+    ...
+  pvc:
+    ...
+  preallocation: true
+```
+
+`preallocation` is a binary option. It defaults to `false`.
+
+## Preallocation for StorageClass
+
+It is also possible to define preallocation for all Data Volumes created with a StorageClass.
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: sc-with-preallocation
+provisioner: ...
+reclaimPolicy: ...
+volumeBindingMode: ...
+preallocation: true
```

Comment:
I don't think we control the fields of storageclass

Reply:
Right we cannot use the storage class for any of this. We should have it in the CDIConfig object at the same level as the fsOverhead. We should probably make it a specific struct for storage class level settings.

Reply:
Because the internal object are de/serialized to YAML, that would mean changing the CDIConfig API. Do we want to do it?